packages/shared/sierra-client/src/HttpSierraClient.ts

```diff
@@ -131,44 +135,160 @@ export default class HttpSierraClient implements SierraClient {
     });
   }
 
-  async updatePatronRecord(
+  async updatePatronEmail(
     recordNumber: number,
-    email: string
+    email: string,
+    verified: boolean = false
   ): Promise<APIResponse<PatronRecord>> {
-    return this.getInstance().then((instance) => {
-      return instance
-        .put(
-          '/patrons/' + recordNumber,
-          {
-            emails: [email],
-          },
-          {
-            validateStatus: (status) => status === 204,
-          }
-        )
-        .then(() => this.getPatronRecordByRecordNumber(recordNumber))
-        .catch((error) => {
-          if (error.response) {
-            switch (error.response.status) {
-              case 400:
-                return errorResponse(
-                  'Malformed or invalid Patron record update request',
-                  ResponseStatus.MalformedRequest,
-                  error
-                );
-              case 404:
-                return errorResponse(
-                  'Patron record with record number [' +
-                    recordNumber +
-                    '] not found',
-                  ResponseStatus.NotFound,
-                  error
-                );
-            }
-          }
-          return unhandledError(error);
-        });
-    });
+    try {
+      const instance = await this.getInstance();
+      const currentRecordResponse = await instance.get<PatronResponse>(
+        `/patrons/${recordNumber}`,
+        {
+          params: { fields: 'varFields' },
+          validateStatus: (status) => status === 200,
+        }
+      );
+
+      const updatedVarFields = verified
+        ? addVerificationNote(currentRecordResponse.data.varFields)
+        : currentRecordResponse.data.varFields;
+
+      await instance.put(
+        '/patrons/' + recordNumber,
+        {
+          emails: [email],
+          varFields: updatedVarFields,
+        },
+        {
+          validateStatus: (status) => status === 204,
+        }
+      );
+
+      return this.getPatronRecordByRecordNumber(recordNumber);
+    } catch (error) {
+      if (error.response) {
+        switch (error.response.status) {
+          case 400:
+            return errorResponse(
+              'Malformed or invalid Patron record update request',
+              ResponseStatus.MalformedRequest,
+              error
+            );
+          case 404:
+            return errorResponse(
+              'Patron record with record number [' +
+                recordNumber +
+                '] not found',
+              ResponseStatus.NotFound,
+              error
+            );
```

Comment:
Not for this patch, but it occurs to me we might want to consolidate some of this logic so we don't have to repeat it on all responses.